thirdeye/thirdeye-frontend/app/pods/rootcause-aggregates-cache/service.js

```diff
@@ -56,12 +56,43 @@ export default Service.extend({
       return;
     }
 
-    // metrics
+    let metricUrnToOffestAndUrn = {};
     missing.forEach(urn => {
-      return this._fetchSlice(urn, requestContext);
+      const metricUrn = toMetricUrn(urn);
+      if (!(metricUrn in metricUrnToOffestAndUrn)) {
+        metricUrnToOffestAndUrn[metricUrn] = [];
+      }
+      metricUrnToOffestAndUrn[metricUrn].push([toAbsoluteUrn(urn, context.compareMode).split(':')[2].toLowerCase(), urn]);
     });
+
+    Object.keys(metricUrnToOffestAndUrn).forEach(
+        metricUrn => {
+          return this._fetchRowSlice(metricUrn, requestContext, metricUrnToOffestAndUrn);
+        });
+  },
+
+  _fetchRowSlice(metricUrn, context, metricUrnToOffestAndUrn){
```

Comment:
Let's document this with jsdoc http://usejsdoc.org/


Here's an example in the code: 
https://github.com/linkedin/pinot/blob/master/thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js#L748